pkgs/build-support/nix-gitignore/default.nix

```diff
@@ -147,10 +147,10 @@ in rec {
         ' sh {} \; > $out
       '');
 
-  withGitignoreFile = patterns: root:
-    lib.toList patterns ++ [(root + "/.gitignore")];
+  withGitIgnoredFilesAndDirs = patterns: root:
```

Comment:
If we rename this, we break the interface I’m not sure this should have been exposed in the first place, so I recommend we move it out of the record.

Reply:
cc @siers

Reply:
I doubt many users are using this function and even if they are, it's easily fixable.

Reply:
You could always add `builtins.trace "please use with..Dirs instead of with...File" with...Dirs`.